### DISCUSSION TITLE
Such a thing as a ButtonField?

### DISCUSSION THREAD

**Discussion by titantwentyone** [filamentphp/filament#1733] [Ideas]
This may be an odd request. I was looking for something that I can drop into a form which invokes a modal and returns a value that can be injected into the state. I can see that we use ButtonActions which are tied to a form but my plan would be to use this anywhere including Builders which is why a ButtonAction wouldn't work for me.

Some context: The idea is to invoke a modal which displays a list of media used in the system. User clicks an image or images, modal is closed and the relevant field is populated with the image ID (path or whatever).

If nothing like this exists, is it worth exploring? Happy to look at a PR.

**Comment by danharrin** [comment]
Sounds like you should just implement a custom media library picker component for your app, which has a button which opens a modal. I don't think I'd merge something like a generic Button component into the Filament core, buttons can look very different between apps.

**Reply by titantwentyone** [reply]
Thank you! I was trying to over-complicate things. Found I can use `<x-filament::modal>` there also.
